File: samples/sample-app-java/src/main/java/com/weedow/spring/data/search/sample/java/entity/Feature.java
package com.weedow.spring.data.search.sample.java.entity;

import javax.persistence.*;
import java.util.Map;

@Entity
public class Feature extends JpaPersistable<Long> {

    @Column(nullable = false, unique = true)
    private String name;

    @Column(nullable = false)
    private String description;

    @ElementCollection
    @CollectionTable(
            name = "metadata_mapping",
            joinColumns = {@JoinColumn(referencedColumnName = "id", name = "feature_id")}
    )
    @MapKeyColumn(name = "metadata_name")
    @Column(name = "value")
    private Map<String, String> metadata;

    public String getName() {
        return name;
    }

    public Feature setName(String name) {
        this.name = name;
        return this;
    }

    public String getDescription() {
        return description;
    }

    public Feature setDescription(String description) {
        this.description = description;
        return this;
    }

    public Map<String, String> getMetadata() {
        return metadata;
    }

    public Feature setMetadata(Map<String, String> metadata) {
        this.metadata = metadata;
        return this;
    }
}

File: common/src/main/kotlin/com/weedow/spring/data/search/common/model/Feature.kt
package com.weedow.spring.data.search.common.model

import javax.persistence.*

@Entity
class Feature(
    @Column(nullable = false, unique = true)
    val name: String,

    @Column(nullable = false)
    val description: String,

    @ElementCollection
    @CollectionTable(
        name = "metadata_mapping",
        joinColumns = [JoinColumn(name = "feature_id", referencedColumnName = "id")]
    )
    @MapKeyColumn(name = "metadata_name")
    @Column(name = "value")
    val metadata: Map<String, String>? = null

) : JpaPersistable<Long>()